src/ctapipe/io/hdf5monitoringsource.py

```
@@ -257,12 +262,25 @@ def __init__(
                     # Overwrite the provided subarray if it is not set
                     if self.subarray is None:
                         self.subarray = subarray_from_file
-                except Exception as e:
-                    self.log.debug(
-                        f"HDF5MonitoringSource: Monitoring file '{file}' does not contain "
-                        f"a subarray description. Reading failed: {e}"
+                except tables.exceptions.NoSuchNodeError as node_missing_error:
+                    # Check if the reading failed with an expected error message or not
+                    if (
+                        str(node_missing_error)
+                        == "group ``/`` does not have a child named ``/configuration/instrument/subarray/layout``"
+                    ):
+                        self.log.debug(
+                            f"HDF5MonitoringSource: Monitoring file '{file}' does not contain a subarray description."
+                        )
+                    else:
+                        raise IOError(
+                            f"HDF5MonitoringSource: Unexpected ``tables.exceptions.NoSuchNodeError`` occurred: {node_missing_error} "
+                            f"Could not read subarray description from monitoring file '{file}'."
+                        )
+                except Exception as unexpected_error:
```

I would not mask any exception as IOError, remove this.